### PR TITLE
fix: update PassageError message precedence

### PR DESCRIPTION
**Pull request by bertrmz**
## What's New?

-   set explicit msg precedence in `PassageError`
-   align get-by-identifier error description with other SDKs
-   use symbols in `PassageError` to support both string and symbol keys
-   update handwritten error codes to match auth OpenAPI schema

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have manually tested my code thoroughly
- [ ] I have added/updated inline documentation for public facing interfaces if relevant
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing integration and unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/passageidentity/error.rb [hunk 1]
```diff
@@ -7,21 +7,45 @@ module Passage
   class PassageError < StandardError
     attr_reader :status_code, :error_code, :message, :status_text, :error
 
-    def initialize(message: nil, status_code: nil, body: nil)
-      super(message)
+    def initialize(message: nil, status_code: nil, body: {})
+      # Ensure body is a hash
+      body = {} if body.nil?
+      body = symbolize_keys(body)
```
Comment (bertrmz) [inline lib/passageidentity/error.rb:13]: The addition here now supports bodies returned by the Passage Auth API, as well as errors filled in with SDK code like: https://github.com/passageidentity/passage-ruby/blob/b4ca0ae675975aa4ecef615214895f9968a059ff/lib/passageidentity/user_api.rb#L51-L57

File: lib/passageidentity/error.rb [hunk 2]
```diff
@@ -7,21 +7,44 @@ module Passage
   class PassageError < StandardError
     attr_reader :status_code, :error_code, :message, :status_text, :error
 
-    def initialize(message: nil, status_code: nil, body: nil)
-      super(message)
+    def initialize(message: nil, status_code: nil, body: {})
+      # Ensure body is a hash
+      body = {} if body.nil?
 
+      body = symbolize_keys(body)
```
Comment (ctran88) [inline lib/passageidentity/error.rb:14]: it looks like this is a [rails method](https://api.rubyonrails.org/classes/Hash.html#method-i-symbolize_keys). will that work in this sdk without us having to explicitly list it in our gemspec?

Comment (bertrmz) [reply lib/passageidentity/error.rb:14]: This invokes a private method: https://github.com/passageidentity/passage-ruby/blob/0a8ed4d94cbb2b532d3b878d6e81ba82a302a619/lib/passageidentity/error.rb#L46-L48 which uses the [transform_keys](https://ruby-doc.org/core-2.5.8/Hash.html#method-i-transform_keys) method of Ruby's Hash class.

Comment (ctran88) [reply lib/passageidentity/error.rb:14]: oh whoops i missed that at the bottom.

**Review by ctran88 — APPROVED** [review]

**Pull request merged**